### PR TITLE
Pay 3955 iac supplementary info

### PR DESCRIPTION
**Pull request by AnilDhawaleGitHub**
**Before creating a pull request make sure that:**

- [ ] commit messages are meaningful and follow good commit message guidelines
- [ ] README and other documentation has been updated / added (if needed)
- [ ] tests have been updated / new tests has been added (if needed)

Please remove this line and everything above and fill the following sections:

### Change description ###

**Does this PR introduce a breaking change?** (check one with "x")

```
[ ] Yes
[ x] No
```

### CODE REVIEW

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/CMCCardPaymentFunctionalTest.java [hunk 1]
```diff
@@ -61,6 +66,10 @@
     private static String SERVICE_TOKEN;
     private static boolean TOKENS_INITIALIZED = false;
 
+    private static final Logger LOG = LoggerFactory.getLogger(CMCCardPaymentFunctionalTest.class);
+
+    private static DateTimeZone zoneUTC = DateTimeZone.UTC;
```
Comment (vigneshpushkaranHMCTS) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/CMCCardPaymentFunctionalTest.java:71]: ZoneUTC variable is unused. Please check.

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java [hunk 2]
```diff
@@ -285,7 +288,7 @@ public void makeAndRetrievePbaPaymentByFinrem() throws InterruptedException {
             .s2sToken(SERVICE_TOKEN)
             .when().searchPaymentsByServiceBetweenDates(Service.FINREM, startDate, endDate)
             .then().getPayments((paymentsResponse -> {
-            Assertions.assertThat(paymentsResponse.getPayments().size()).isEqualTo(1);
+            Assertions.assertThat(paymentsResponse.getPayments().size()).isGreaterThanOrEqualTo(1);
```
Comment (vigneshpushkaranHMCTS) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java:291]: why isGreaterThanOrEqualTo needs here?

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java [hunk 3]
```diff
@@ -307,7 +310,7 @@ public void makeAndRetrievePbaPaymentByCivilService() throws InterruptedExceptio
             .s2sToken(SERVICE_TOKEN)
             .when().searchPaymentsByServiceBetweenDates(Service.CIVIL, startDate, endDate)
             .then().getPayments((paymentsResponse -> {
-            Assertions.assertThat(paymentsResponse.getPayments().size()).isEqualTo(1);
+            Assertions.assertThat(paymentsResponse.getPayments().size()).isGreaterThanOrEqualTo(1);
```
Comment (vigneshpushkaranHMCTS) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java:313]: why isGreaterThanOrEqualTo needs here?

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java [hunk 4]
```diff
@@ -330,7 +333,7 @@ public void makeAndRetrievePbaPaymentByIACService() throws InterruptedException
             .s2sToken(SERVICE_TOKEN)
             .when().searchPaymentsByServiceBetweenDates(Service.IAC, startDate, endDate)
             .then().getPayments((paymentsResponse -> {
-            Assertions.assertThat(paymentsResponse.getPayments().size()).isEqualTo(1);
+            Assertions.assertThat(paymentsResponse.getPayments().size()).isGreaterThanOrEqualTo(1);
```
Comment (vigneshpushkaranHMCTS) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java:336]: why isGreaterThanOrEqualTo needs here?

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java [hunk 5]
```diff
@@ -353,7 +356,7 @@ public void makeAndRetrievePbaPaymentByFPLService() throws InterruptedException
             .s2sToken(SERVICE_TOKEN)
             .when().searchPaymentsByServiceBetweenDates(Service.FPL, startDate, endDate)
             .then().getPayments((paymentsResponse -> {
-            Assertions.assertThat(paymentsResponse.getPayments().size()).isEqualTo(1);
+            Assertions.assertThat(paymentsResponse.getPayments().size()).isGreaterThanOrEqualTo(1);
```
Comment (vigneshpushkaranHMCTS) [inline api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java:359]: why isGreaterThanOrEqualTo needs here?

File: api/src/main/java/uk/gov/hmcts/payment/api/configuration/RestTemplateConfiguration.java [hunk 6]
```diff
@@ -20,4 +20,12 @@ public CloseableHttpClient paymentsHttpClient() {
     public RestTemplate restTemplatePaymentGroup() {
         return new RestTemplate(new HttpComponentsClientHttpRequestFactory());
     }
+
+    @Bean (value = "restTemplateIacSupplementaryInfo")
+    public RestTemplate restTemplateIacSupplementaryInfo() {
+        var factory = new HttpComponentsClientHttpRequestFactory();
+        factory.setConnectTimeout(5000);
```
Comment (vigneshpushkaranHMCTS) [inline api/src/main/java/uk/gov/hmcts/payment/api/configuration/RestTemplateConfiguration.java:27]: Can we configure in properties file for value?

File: api/src/main/resources/application-local.properties [hunk 7]
```diff
@@ -71,9 +72,9 @@ management.endpoints.web.base-path=/
 management.endpoints.web.exposure.include=health,info
 management.endpoint.health.show-details=always
 
-fees.register.url=https://fees-register-api.platform.hmcts.net
+fees.register.url=http://localhost:8091
 
-feature.payments.search=false
+feature.payments.search=true
```
Comment (vigneshpushkaranHMCTS) [inline api/src/main/resources/application-local.properties:77]: revert to old config

File: api/src/main/resources/application-local.properties [hunk 8]
```diff
@@ -71,9 +72,9 @@ management.endpoints.web.base-path=/
 management.endpoints.web.exposure.include=health,info
 management.endpoint.health.show-details=always
 
-fees.register.url=https://fees-register-api.platform.hmcts.net
+fees.register.url=http://localhost:8091
```
Comment (vigneshpushkaranHMCTS) [inline api/src/main/resources/application-local.properties:75]: revert to old config

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/CMCCardPaymentFunctionalTest.java (hunk 1 above)
Comment (AnilDhawaleGitHub) [reply api/src/functionalTest/java/uk/gov/hmcts/payment/functional/CMCCardPaymentFunctionalTest.java]: Fixed it

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java (hunk 5 above)
Comment (AnilDhawaleGitHub) [reply api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java]: resolved

File: api/src/main/java/uk/gov/hmcts/payment/api/configuration/RestTemplateConfiguration.java (hunk 6 above)
Comment (AnilDhawaleGitHub) [reply api/src/main/java/uk/gov/hmcts/payment/api/configuration/RestTemplateConfiguration.java]: resolved

File: api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java (hunk 2 above)
Comment (AnilDhawaleGitHub) [reply api/src/functionalTest/java/uk/gov/hmcts/payment/functional/PBAPaymentFunctionalTest.java]: Resolved

File: api/src/main/resources/application-local.properties (hunk 7 above)
Comment (AnilDhawaleGitHub) [reply api/src/main/resources/application-local.properties]: resolved

File: api/src/main/resources/application-local.properties (hunk 8 above)
Comment (AnilDhawaleGitHub) [reply api/src/main/resources/application-local.properties]: Resolved

**Review by vigneshpushkaranHMCTS — APPROVED** [review]

File: model/src/main/java/uk/gov/hmcts/payment/api/dto/IacSupplementaryRequest.java [hunk 9]
```diff
@@ -0,0 +1,24 @@
...
+import lombok.*;
+import lombok.experimental.Wither;
+import javax.validation.Valid;
+import java.util.List;
+import static com.fasterxml.jackson.annotation.JsonInclude.Include.NON_NULL;
+
+@With
+@JsonNaming(PropertyNamingStrategy.SnakeCaseStrategy.class)
+@JsonInclude(NON_NULL)
+@Data
```
[the remark this answers is not in the document]
Comment (AnilDhawaleGitHub) [reply model/src/main/java/uk/gov/hmcts/payment/api/dto/IacSupplementaryRequest.java]: done

File: api/src/main/java/uk/gov/hmcts/payment/api/service/IacServiceImpl.java [hunk 10]
```diff
@@ -0,0 +1,117 @@
...
+            } catch (HttpClientErrorException ex) {
+                LOG.info("IAC Supplementary information could not be found, exception: {}", ex.getMessage());
+                paymentResponseHttpStatus = HttpStatus.PARTIAL_CONTENT;
+                isExceptionOccur = true;
+            } catch (Exception ex) {
+                LOG.info("Unable to retrieve IAC Supplementary Info information, exception: {}", ex.getMessage());
+                paymentResponseHttpStatus = HttpStatus.PARTIAL_CONTENT;
+                isExceptionOccur = true;
+            }
+            if (!isExceptionOccur) {
```
Comment (aravindan89) [inline api/src/main/java/uk/gov/hmcts/payment/api/service/IacServiceImpl.java:67]: @AnilDhawaleGitHub - Do we need this boolean variable explicitly ?

**Review by krishnanuthalapati — APPROVED** [review]

**Pull request merged**